### PR TITLE
Keybinds

### PR DESCRIPTION
**Pull request by Incompleteusern**

### CODE REVIEW

**Comment by Incompleteusern** [comment]
fabric api does not exist, going to have to deal with that :(

**Comment by magicus** [comment]
I actually wonder how hard it would be to get fabric api working on forge...

**Comment by Incompleteusern** [comment]
Can't you just put the fabric api in the relevant `build.gradle`? It should work now, and Im going to try and clean up the commit history

**Comment by Incompleteusern** [comment]
The event bus transformer isn't applied to the tick event, so fabric api likely needs to be used. Im going to use fabric api until a good alternative solution is found

**Comment by Incompleteusern** [comment]
Anyways, this should be the final commit (fingers crossed). Here's some images
image
image
image

File: common/src/main/java/com/wynntils/mc/mixin/KeyMappingMixin.java [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+public class KeyMappingMixin {
+    @Shadow @Final private static Map<String, Integer> CATEGORY_SORT_ORDER;
+
+    @Inject(
+            method =
+                    "<init>(Ljava/lang/String;Lcom/mojang/blaze3d/platform/InputConstants$Type;ILjava/lang/String;)V",
+            at = @At("RETURN"))
+    public void initPost(
+            String name, InputConstants.Type type, int i, String category, CallbackInfo ci) {
+        if (CATEGORY_SORT_ORDER.containsKey(category)) return;
```
Comment (magicus) [inline common/src/main/java/com/wynntils/mc/mixin/KeyMappingMixin.java:27]: @Incompleteusern Sorry for not reviewing this earlier. Can you extract this code in the external class like other mixins, and just pass the CATEGORY_SORT_ORDER as argument? Code in mixins is hard to debug. You can't put breakpoints or do hot swapping, so the idea is to make the actual mixin as minimal as possible, just a hook.

File: common/src/main/java/com/wynntils/utils/Utils.java [hunk 2]
```diff
@@ -4,24 +4,8 @@
...
 
-import com.wynntils.core.WynntilsMod;
-import com.wynntils.wc.Models;
-import net.minecraftforge.eventbus.api.IEventBus;
-
 /**
  * This is a "high-quality misc" class. Helper methods that are commonly used throughout the project
  * without an aspect on minecraft can be put here. Keep the names short, but distinct.
  */
-public class Utils {
```
Comment (magicus) [inline common/src/main/java/com/wynntils/utils/Utils.java:15]: @Incompleteusern You did not think we should have a generic Util class for a few select methods?

File: common/src/main/resources/wynntils.mixins.json [hunk 3]
```diff
@@ -8,11 +8,14 @@
...
     "ConnectionMixin",
+    "ConnectScreenMixin",
     "MinecraftMixin",
     "MultiPlayerGameModeMixin",
+    "OptionsMixin",
     "ScoreboardMixin",
     "ScreenMixin",
-    "ConnectScreenMixin"
+    "KeyMappingMixin",
+    "accessors.OptionsAccessor"
```
Comment (magicus) [inline common/src/main/resources/wynntils.mixins.json:18]: @Incompleteusern To keep this list sane, I've tried to keep it sorted alphabetically. I see you fixed ConnectScreenMixin, but KeyMappingMixin ended up last...

Comment (Incompleteusern) [reply common/src/main/resources/wynntils.mixins.json:18]: Okay, Ill probably fix that later

File: common/src/main/java/com/wynntils/utils/Utils.java (hunk 2 above)
Comment (Incompleteusern) [reply common/src/main/java/com/wynntils/utils/Utils.java:15]: The thing is both of those methods have better places to be, or in the case of the first one is unnecessary

File: common/src/main/java/com/wynntils/mc/mixin/KeyMappingMixin.java (hunk 1 above)
Comment (Incompleteusern) [reply common/src/main/java/com/wynntils/mc/mixin/KeyMappingMixin.java:27]: This code should be relatively invariable so it is likely fine to stay as is, furthermore, I don't see this changing in the future much

Comment (Incompleteusern) [reply common/src/main/java/com/wynntils/mc/mixin/KeyMappingMixin.java:27]: Feel free to extract it though

File: common/src/main/resources/wynntils.mixins.json (hunk 3 above)
Comment (Incompleteusern) [reply common/src/main/resources/wynntils.mixins.json:18]: Done

**Pull request merged**